### ISSUE TITLE
timestamping crashing 1.0.39

### ISSUE THREAD

**Issue by si458** [Ylianst/MeshCentral#4161]
**Describe the bug**
```
MeshCentral HTTP redirection server running on port 80.
MeshCentral v1.0.39, Hybrid (LAN + WAN) mode, Production mode.
ERR: /opt/meshcentral/meshcentral/authenticode.js:1301
            pkcs7der.value[1].value[0].value[2].value[0].value = forge.asn1.oidToDer(forge.pki.oids.data).data;
                     ^

TypeError: Cannot read properties of null (reading 'value')
    at obj.sign (/opt/meshcentral/meshcentral/authenticode.js:1301:22)
    at CreateMeshCentralServer.obj.signMeshAgents (/opt/meshcentral/meshcentral/meshcentral.js:2979:39)
    at CreateMeshCentralServer.obj.StartEx4 (/opt/meshcentral/meshcentral/meshcentral.js:1620:13)
    at CreateMeshCentralServer.obj.StartEx3 (/opt/meshcentral/meshcentral/meshcentral.js:1568:17)
    at /opt/meshcentral/meshcentral/meshcentral.js:1480:21
    at obj.GetMeshServerCertificate (/opt/meshcentral/meshcentral/certoperations.js:986:43)
    at CreateMeshCentralServer.obj.StartEx2 (/opt/meshcentral/meshcentral/meshcentral.js:1477:35)
    at Server.<anonymous> (/opt/meshcentral/meshcentral/redirserver.js:143:13)
    at Object.onceWrapper (node:events:641:28)
    at Server.emit (node:events:527:28)

Node.js v18.3.0
Error: Command failed: /usr/local/bin/node /opt/meshcentral/meshcentral/meshcentral --launch 7
/opt/meshcentral/meshcentral/authenticode.js:1301
            pkcs7der.value[1].value[0].value[2].value[0].value = forge.asn1.oidToDer(forge.pki.oids.data).data;
                     ^

TypeError: Cannot read properties of null (reading 'value')
    at obj.sign (/opt/meshcentral/meshcentral/authenticode.js:1301:22)
    at CreateMeshCentralServer.obj.signMeshAgents (/opt/meshcentral/meshcentral/meshcentral.js:2979:39)
    at CreateMeshCentralServer.obj.StartEx4 (/opt/meshcentral/meshcentral/meshcentral.js:1620:13)
    at CreateMeshCentralServer.obj.StartEx3 (/opt/meshcentral/meshcentral/meshcentral.js:1568:17)
    at /opt/meshcentral/meshcentral/meshcentral.js:1480:21
    at obj.GetMeshServerCertificate (/opt/meshcentral/meshcentral/certoperations.js:986:43)
    at CreateMeshCentralServer.obj.StartEx2 (/opt/meshcentral/meshcentral/meshcentral.js:1477:35)
    at Server.<anonymous> (/opt/meshcentral/meshcentral/redirserver.js:143:13)
    at Object.onceWrapper (node:events:641:28)
    at Server.emit (node:events:527:28)

Node.js v18.3.0

    at ChildProcess.exithandler (node:child_process:387:12)
    at ChildProcess.emit (node:events:539:35)
    at maybeClose (node:internal/child_process:1090:16)
    at Socket.<anonymous> (node:internal/child_process:449:11)
    at Socket.emit (node:events:527:28)
    at Pipe.<anonymous> (node:net:740:14) {
  code: 1,
  killed: false,
  signal: null,
  cmd: '/usr/local/bin/node /opt/meshcentral/meshcentral/meshcentral --launch 7'
}
ERROR: MeshCentral failed with critical error, check mesherrors.txt. Restarting in 5 seconds...
```

**To Reproduce**
1. deleted signedagents folder
2. updated docker image to latest npm
3. start docker

**Expected behavior**
it to work?

**Screenshots**

**Server Software (please complete the following information):**
 - OS: [e.g. Ubuntu] Alpine
 - Virtualization: [e.g. Docker] Docker
 - Network: [e.g. LAN/WAN, reverse proxy, cloudflare, ssl offload, etc...] Hybrid Mode, Traefik Reverse Proxy
 - Version: [e.g. 1.0.36] 1.0.39
 - Node: [e.g. 18.3.0]
 - Browser: [e.g. Google Chrome] Firefox

**Remote Device (please complete the following information):**
 - Device: [e.g. Laptop]
 - OS: [e.g. Windows 10]
 - Version: [e.g. 21H2]
 - Current Core Version (if known): [**HINT**: Go to a device then `console` Tab then type `info`]

**Additional context**

**Comment by si458** [comment]
found the bug - you need to include inside your config.json at least one entry of `agentFileInfo` like `fileDescription`
othewise it just crashes
https://github.com/Ylianst/MeshCentral/blob/master/meshcentral-config-schema.json#L546-L559

**Comment by si458** [comment]
found the bug again and narrowed it down if you include a FULL STOP in any of the entries it will crash
hense the default `"legalCopyright": "Apache 2.0 License"`

**Comment by silversword411** [comment]
Good catch!

> change it to `"legalCopyright": "Apache 20 License"` and it works a treat

Probably `"legalCopyright": "Apache 2 License"` is more appropriate ;)

**Comment by Ylianst** [comment]
Oh dear! Looking into this now.

**Comment by Ylianst** [comment]
Just published v1.0.40 with many bug fixes to agent signing. Some things where broken, other problems also occurred with older versions of NodeJS. Let me know if it works now.

**Comment by si458** [comment]
> Just published v1.0.40 with many bug fixes to agent signing. Some things where broken, other problems also occurred with older versions of NodeJS. Let me know if it works now.

BINGO! pulled latest git, built docker, stopped/started docker-compose, works a treat now! thanks!

**Comment by Ylianst** [comment]
Thank you for making such outstanding bug reports!!! I very much appreciate the help.
